github/transport.go

```diff
@@ -197,3 +197,104 @@ func isWriteMethod(method string) bool {
 	}
 	return false
 }
+
+type RetryTransport struct {
+	transport       http.RoundTripper
+	retryDelay      time.Duration
+	maxRetries      int
+	retryableErrors map[int]bool
+}
+
+type RetryTransportOption func(*RetryTransport)
+
+// NewRetryTransport takes in an http.RoundTripper and a variadic list of
+// optional functions that modify the RetryTransport struct itself. This
+// may be used to retry after response errors 5xx, for example.
+func NewRetryTransport(rt http.RoundTripper, options ...RetryTransportOption) *RetryTransport {
+	// Default to no retry if none is provided
+	defaultErrors := map[int]bool{
+		500: true,
```

Comment:
One more note here: I'm a little concerned that we're defining the default HTTP errors to retry twice: once here in transport.go and again on line 378 on provider.go. 

Can we consolidate that duplication?

Reply:
Sure! Added a function on repository_utils.go, not sure if it is the beast tho